transcom/mymove
# B-21375 I-13559 Calculated UB allowance when either dependents age group is 0 - INT

## [Agility issue](https://www13.v1host.com/USTRANSCOM38/Issue.mvc/Summary?oidToken=Issue%3A1034479)
## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21375)

## Summary

Added a fix here to calculate the civilian UB allowance for the case of 0 for either the dependents_under_12 or dependents_12_and_over values. Had a logical AND (&&) check that required both values to be non-zero for the UB allowance to be calculated - replaced that with a logical OR (||) check instead.

To test:
1. Log in as a customer
2. Have your UB FF on
3. At the point of uploading orders, select an international duty location
4. Select yes for dependents, and have one of the age groups be 0 (either under 12 or 12 and over), and one be a value greater than 0.
5. For grade, select Civilian (this step is important)
![image](https://github.com/user-attachments/assets/51bd6945-2b22-4a0f-ad61-e3aedc0cff95)

6. Complete the orders upload step
7. Click on the set up your shipments button
8. Check on the next page that you see a UB allowance shown and that it matches the logic for civilian UB allowances, here are the values to help you do some math:
```
const civilianBaseUBAllowance = 350
const dependents12AndOverUBAllowance = 350
const depedentsUnder12UBAllowance = 175
const maxWholeFamilyCivilianUBAllowance = 2000

```

I see 1,050 for the UB allowance, based on 350 for the base civilian allowance + 2 * 350 for each of the dependents over 12
![image](https://github.com/user-attachments/assets/1dff87bb-5cfa-48be-a313-847723a6ae3b)
